### ISSUE TITLE
Schematron rule TAF.TAF18 mandates the presence of iwxxm:cloud which should not be the case for CAVOK

### ISSUE THREAD

**Issue by blchoy** [wmo-im/iwxxm#24]
Jean-François Menon of Meteo France reported that TAF.TAF18 mandates the presence of iwxxm:surfaceWind and iwxxm:cloud when iwxxm:baseForecast is present. For TAC, however, the cloud group should not be present when CAVOK is expected. To mimic this arrangement TAF.TAF18 should be revised.

**Comment by marqh** [comment]
updates required in:
* [ ] UML master Model in Enterprise Architect, generating:
  * [ ] HTML files generated from the model
  * [ ] schematron on schemas.wmo.int:
    * http://schemas.wmo.int/index.php?dir=/iwxxm/2.1/zip

**Comment by marqh** [comment]
version labelling:  
* http://schemas.wmo.int/index.php?dir=/iwxxm
  * 2.1.1
    * fully backwards compatible version with only editorial changes and fixes to identified bugs

**Comment by blchoy** [comment]
TAF.TAF18 now split up into TAF.TAF18 and TAF.TAF19.  Changes (together with slight enhancement to other schematron rules for TAF) have been uploaded to the repository for checking and further comments.

**Comment by braeckel** [comment]
Several of the TAF changes you've checked in seem significant and should probably be tested against examples.  I think that is the only way to verify that they are correct.

**Comment by braeckel** [comment]
Were these changes generated from EA or manually edited?  If they weren't generated from EA we need to ensure that the export process exactly matches the current checked-in files.

**Comment by blchoy** [comment]
The *.sch files are generated from the UML.  You can find the XMI in the ProjectLocker SVN.  I agree with Aaron that changes to the schematron rules need through checking, and that's why I didn't close this ticket after submitting the changes.  :)

**Comment by braeckel** [comment]
Using the new checked-in rules it is now possible to correctly report CAVOK situations like the following and fail in other cases when CAVOK is true and clouds, presentWeather, and so on are present.  Therefore the updated Schematron rules appear to be correct for METAR.  All existing examples validate against the updated Schematron rules, so everything seems to be working and we can close this issue.

```XML
<iwxxm:MeteorologicalAerodromeObservationRecord gml:id="or1" cloudAndVisibilityOK="true">
      <iwxxm:airTemperature uom="Cel">17.0</iwxxm:airTemperature>
      <iwxxm:dewpointTemperature uom="Cel">16.0</iwxxm:dewpointTemperature>
      <iwxxm:qnh uom="hPa">1018</iwxxm:qnh>
      <iwxxm:surfaceWind>
          <iwxxm:AerodromeSurfaceWind variableWindDirection="false">
              <iwxxm:meanWindDirection uom="deg">240</iwxxm:meanWindDirection>
              <iwxxm:meanWindSpeed uom="m/s">4.0</iwxxm:meanWindSpeed>
          </iwxxm:AerodromeSurfaceWind>
      </iwxxm:surfaceWind>
</iwxxm:MeteorologicalAerodromeObservationRecord>
```

**Comment by braeckel** [comment]
As a note, the commit that fixed this issue is cf67be25ec5ee15bf1a64298032669b741532abc
